src/components/Form/SignInForm/SignInForm.js

```diff
@@ -0,0 +1,80 @@
+import React, { useState, useRef, useEffect } from 'react';
+import Button from '@material-ui/core/Button';
+import { ValidatorForm, TextValidator } from 'react-material-ui-form-validator';
+import { serviceRequest } from '../../../services/serviceRequest';
+import config from '../../../config';
+
+const emailPattern = /^[A-Za-z0-9_!#$%&'*+=?`{|}~^.-]+@[A-Za-z0-9_!#$%&'*+=?`{|}~^.-]+.edu$/;
+
+export const SignInForm = () => {
+  const [formData, setForm] = useState({ email: '', password: '' });
+  const inputRef = useRef(null);
+
+  useEffect(() => {
+    ValidatorForm.addValidationRule('isValidEmail', (email) => {
+      if (email) { return emailPattern.test(email); }
+      return true;
+    });
+  });
+
+  function handleChange(event) {
+    const updateForm = { ...formData };
+    updateForm[event.target.name] = event.target.value;
+    setForm(updateForm);
+  }
+
+  async function handleSubmit() {
+    try {
+      const uri = `${config.api_domain}/api/v1/signin`;
+      const response = await serviceRequest({
```

Comment:
- Beautiful abstractions of request, this is what we wanted to see and read. You can even create an auxiliary function to prepare your request payload. 
```js
const preparePayload = (...) => {
  ...
}
```
- This reduces the the line of code and enhances readability. _**(This is optional, what you have already looks amazing)**_

Reply:
sure